src/regression.js

```diff
@@ -136,6 +150,8 @@
       var gradient;
       var intercept;
       var len = data.length;
+      var nonNullLen = 0; // the number of non-null points
```

Comment:
Seemed unnecessary to add another loop, especially considering that I've had to add the loop to the r2 (which itself contains several loops)